plugins/reporters/opossum/src/main/kotlin/OpossumReporter.kt

```diff
@@ -96,50 +104,78 @@ class OpossumReporter(
-        val uuid: UUID = UUID.randomUUID()
+        @Transient
+        val uuid: UUID = UUID.randomUUID(),
+        val source: OpossumSignalSource,
+        val attributionConfidence: Int = 80,
+        val packageType: String?,
+        val packageNamespace: String?,
+        val packageName: String?,
+        val packageVersion: String?,
+        val copyright: String?,
+        val licenseName: String?,
+        val url: String?,
+        val preSelected: Boolean,
+        val followUp: OpossumFollowUp?,
+        val excludeFromNotice: Boolean,
+        val comment: String?
     ) {
-        fun toJson(): Map<*, *> =
-            sortedMapOf(
-                uuid.toString() to sortedMapOf(
-                    "source" to sortedMapOf(
-                        "name" to source,
-                        "documentConfidence" to 80
-                    ),
-                    "attributionConfidence" to 80,
-                    "packageType" to id?.getPurlType(),
-                    "packageNamespace" to id?.namespace,
-                    "packageName" to id?.name,
-                    "packageVersion" to id?.version,
-                    "copyright" to copyright,
-                    "licenseName" to license?.toString(),
-                    "url" to url,
-                    "preSelected" to preselected,
-                    "followUp" to "FOLLOW_UP".takeIf { followUp },
-                    "excludeFromNotice" to excludeFromNotice,
-                    "comment" to comment
+        companion object {
+            fun create(
+                source: String,
+                id: Identifier? = null,
+                url: String? = null,
+                license: SpdxExpression? = null,
+                copyright: String? = null,
+                comment: String? = null,
+                preSelected: Boolean = false,
+                followUp: Boolean = false,
+                excludeFromNotice: Boolean = false
+            ): OpossumSignal {
+                return OpossumSignal(
+                    source = OpossumSignalSource(name = source),
+                    packageType = id?.getPurlType(),
+                    packageNamespace = id?.namespace,
+                    packageName = id?.name,
+                    packageVersion = id?.version,
+                    copyright = copyright,
+                    licenseName = license?.toString(),
+                    url = url,
+                    preSelected = preSelected,
+                    followUp = OpossumFollowUp.FOLLOW_UP.takeIf { followUp },
+                    excludeFromNotice = excludeFromNotice,
+                    comment = comment,
                 )
-            )
+            }
+        }
 
         fun matches(other: OpossumSignal): Boolean =
             source == other.source
-                && id == other.id
-                && url == other.url
-                && license == other.license
+                && packageType == other.packageType
```

Comment:
Seems like a good idea to me. The `matches` function is not great. Then, we would either need an additional data class for serialization in which all properties are top level again or we need a custom serializer. If that is acceptable, I would rather go with an additional class. What do you think?

Reply:
Why would we even need a custom serialized? From the compared properties, all but `source: OpossumSignalSource` are primitives, and `OpossumSignalSource` itself also only contains primitives. So I believe that new class would not require a customer serializer (not the class using that new class).

Reply:
Not sure I understood you correctly. May be check the code, to see if I got you right. I went with the approach of an additional `OpossumSignalFlat` class instead of a serializer.